### PR TITLE
INT-4036 - Move `@jupiterone/snyk-client` code into `graph-snyk`

### PR DESCRIPTION
**Pull request by austinkelleher**
The decoupled packages make it difficult to quickly add new features
to graph-snyk. Additionally, graph-snyk does not use all of the
functionality of `@jupiterone/snyk-client`.

### CODE REVIEW

File: test/recording.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  SetupRecordingInput,
+} from '@jupiterone/integration-sdk-testing';
+
+type WithRecordingParams = {
+  recordingName: string;
+  directoryName: string;
+  recordingSetupOptions?: SetupRecordingInput['options'];
+};
+
+async function withRecording(
```
Comment (austinkelleher) [inline test/recording.ts:12]: I think we should move this function into `@jupiterone/integration-sdk-testing` and support passing a custom `setupRecording` function optionally. It's a lot cleaner than adding an `afterEach` to every test suite and keeps `recording` out of the global scope in test files.

FYI @VDubber

**Review by mdaum — APPROVED** [review]

File: test/recording.ts (hunk 1 above)
Comment (VDubber) [reply test/recording.ts:12]: It does make the test a bit more dense to read (nested code). However, I think it's a step in the right direction.
What if we extend jest.test as `testWithRecording`... Something like:

```typescript
testWithRecording(
  'test description goes here',
  {
    directoryName: __dirname,
    recordingName: 'fetch-projects',
  }, 
  async () => {
     // Test code goes here
  }
)
```

File: package.json [hunk 2]
```diff
@@ -33,9 +33,11 @@
     "@jupiterone/integration-sdk-core": "^8.13.11",
     "@jupiterone/integration-sdk-dev-tools": "^8.13.11",
     "@jupiterone/integration-sdk-testing": "^8.13.11",
+    "@types/node-fetch": "^2.6.1",
     "ts-node": "^8.10.2"
   },
   "dependencies": {
-    "@jupiterone/snyk-client": "^1.1.2"
+    "@lifeomic/attempt": "^3.0.3",
+    "node-fetch": "^2"
```
Comment (VDubber) [inline package.json:41]: sdk http client still not what you need?

File: src/snyk/client.ts [hunk 3]
```diff
@@ -3,9 +3,10 @@ import {
   IntegrationProviderAPIError,
   IntegrationProviderAuthenticationError,
 } from '@jupiterone/integration-sdk-core';
-import SnykClient from '@jupiterone/snyk-client';
```
Comment (VDubber) [inline src/snyk/client.ts:6]: Was the SnykClient package being used elsewhere?

**Review by VDubber — APPROVED** [review]
Looks good! Minor comments added

File: package.json (hunk 2 above)
Comment (austinkelleher) [reply package.json:41]: Can't really adopt until we have better support for error handling I think. Some status info isn't available. Some addressed here I think: https://github.com/JupiterOne/sdk/pull/723

**Review by VDubber — APPROVED** [review]

**Pull request merged**